### PR TITLE
feat(Modal): handle onClose and show logic of component

### PR DESCRIPTION
**Pull request by YossiSaadi**
### `onClose` and `show` logic

- Also fix types for `Modal` and `ModalTopActions` to be more specific for each use case

https://monday.monday.com/boards/3532715121/pulses/7368472599

### CODE REVIEW

File: packages/core/src/components/ModalNew/ModalTopActions/ModalTopActions.types.ts [hunk 1]
```diff
@@ -13,5 +13,5 @@ export interface ModalTopActionsProps {
   renderAction?: (color?: ModalTopActionsButtonColor) => React.ReactElement<typeof MenuButton | typeof IconButton>;
   color?: ModalTopActionsColor;
   closeButtonAriaLabel?: string;
-  onClose?: React.MouseEventHandler<HTMLDivElement>;
+  onClose?: (e: React.MouseEvent<HTMLButtonElement>) => void;
```
Comment (talkor) [inline packages/core/src/components/ModalNew/ModalTopActions/ModalTopActions.types.ts:16]: Nitpick: Why just e and not event?

File: packages/core/src/components/ModalNew/Modal/Modal.tsx [hunk 2]
```diff
@@ -43,27 +42,57 @@ const Modal = forwardRef(
...
+        if (!show) return;
+        onClose(e);
+      },
+      [onClose, show]
+    );
+
+    useKeyEvent({
+      callback: onEscClick,
+      capture: true,
+      keys: ["Escape"]
```
Comment (talkor) [inline packages/core/src/components/ModalNew/Modal/Modal.tsx:64]: I think we have consts for these

File: packages/core/src/components/ModalNew/Modal/__tests__/Modal.test.tsx [hunk 3]
```diff
@@ -27,6 +28,16 @@ describe("Modal", () => {
     expect(getByTestId("modal")).toHaveAttribute("aria-modal", "true");
   });
 
+  it("does not render when 'show' is false", () => {
```
Comment (talkor) [inline packages/core/src/components/ModalNew/Modal/__tests__/Modal.test.tsx:31]:
```suggestion
  it("should not render when 'show' is false", () => {
```

**Review by talkor — APPROVED** [review]

File: packages/core/src/components/ModalNew/Modal/__tests__/Modal.test.tsx (hunk 3 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/Modal/__tests__/Modal.test.tsx:31]: I've wrote all the tests not in a `should` style (by mistake, accidentally ignored Gebeta's chosen style)
planning on changing all the tests names later 😓

File: packages/core/src/components/ModalNew/ModalTopActions/ModalTopActions.types.ts (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/ModalTopActions/ModalTopActions.types.ts:16]: hmm no special reason
will change it

File: packages/core/src/components/ModalNew/Modal/Modal.tsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/Modal/Modal.tsx:64]: Will change for now, but I think we should make `useKeyEvent` use string keys as well, same as we switched from static props to inline string props

can be a non-breaking change after we release Vibe3

Comment (talkor) [reply packages/core/src/components/ModalNew/Modal/Modal.tsx:64]: But it's not type-safe, no?

Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/Modal/Modal.tsx:64]: it can be
we can extract the keys strings from React's TS I believe

**Pull request merged**